kotlin_spark/src/main/java/com/sleazyweasel/eboshi/AccountRoutes.kt

```diff
@@ -0,0 +1,49 @@
+package com.sleazyweasel.eboshi
+
+import org.eclipse.jetty.http.HttpStatus
+import org.jose4j.jwt.consumer.InvalidJwtException
+import spark.Request
+import spark.Response
+import javax.inject.Inject
+
+class AccountRoutes @Inject constructor(accountDataAccess: AccountDataAccess, auth: Auth) {
+    val create = { input: JsonApiRequest, response: Response ->
+        val submittedUserAttributes = input.data.attributes
+        val user = Account(submittedUserAttributes)
+        val (hashed, salt) = auth.encrypt(user.password!!)
```

Comment:
it forces a possibly null value to be non-null.  It will throw an exception if `user.password` is null.  the `auth.encrypt` function requires a non-null value, so we have to force it in.  and, if you don't put in a password, I'd expect bad things to happen.  In a real-life API impl, we'd want to catch the exception and return a pretty response to the user (400 of some sort, or maybe 401?).  But, since it's not in the test cases, I didn't implement that. :)